pkg/processor/reportprocessor.go

```diff
@@ -191,68 +191,105 @@ func getGovernanceResults(dynamicClient dynamic.Interface, clusterInfo types.Man
+		glog.Warningf("error parsing metadata as a map for policy %s: %s", plcName, err)
+		status, _, err := unstructured.NestedMap(plc.Object, "status")
+		glog.Warningf("error parsing status as a map for policy %s: %s", plcName, err)
+		details, _, err := unstructured.NestedSlice(status, "details")
+		glog.V(1).Infof("error parsing status details as a map for policy %s: %s", plcName, err)
+
+		// Find non-compliant policies
+		if status["compliant"].(string) == "NonCompliant" {
+			// Generate a cluster policy report result for each non-compliant policy within the root policy
+			for idx, detail := range details {
+				var detailMap map[string]interface{}
+				var ok bool
+				if detailMap, ok = detail.(map[string]interface{}); !ok {
+					glog.Warningf("failed to parse history detail as a map for policy %s, template %d", plcName, idx)
+
+					continue
 				}
-			}()
-
-			md := plc.Object["metadata"].(map[string]interface{})
-			status := plc.Object["status"].(map[string]interface{})
-			details := status["details"].([]interface{})
-			if status["compliant"].(string) == "NonCompliant" {
-				for _, detail := range details {
-					if detail.(map[string]interface{})["compliant"] == "NonCompliant" {
-						templateMeta := detail.(map[string]interface{})["templateMeta"].(map[string]interface{})
-						historyItem := detail.(map[string]interface{})["history"].([]interface{})[0].(map[string]interface{})
-						annotations := md["annotations"].(map[string]interface{})
-						category := ""
-						if _, ok := annotations["policy.open-cluster-management.io/categories"]; ok {
-							category = annotations["policy.open-cluster-management.io/categories"].(string)
-						}
-						clusterViolations = append(clusterViolations, &v1alpha2.PolicyReportResult{
-							Policy:      md["name"].(string),
-							Description: historyItem["message"].(string),
-							Scored:      false,
-							Category:    category,
-							Source:      "grc",
-							Timestamp:   metav1.Timestamp{Seconds: time.Now().Unix(), Nanos: int32(time.Now().UnixNano())},
-							Result:      "fail",
-							Properties: map[string]string{
-								"created_at": md["creationTimestamp"].(string),
-								"total_risk": convertSevFromGovernance(getSevFromTemplate(plc, templateMeta["name"].(string))),
-							},
-						})
-					}
+
+				// Find non-compliant wrapped policy templates
+				if detailMap["compliant"] == "NonCompliant" {
+					plcTemplateName, _, err := unstructured.NestedString(detailMap, "templateMeta", "name")
+					glog.Warningf(
+						"error parsing policy template name as a string for policy %s, template %d: %s", plcName, idx, err)
+					historyItems, _, err := unstructured.NestedSlice(detailMap, "history")
+					glog.V(1).Infof("error parsing history as a slice for policy %s, template %d: %s", plcName, idx, err)
+					message, _, err := unstructured.NestedString(historyItems[0].(map[string]interface{}), "message")
+					glog.V(1).Infof(
```

Comment:
Should there be `if err != nil` conditionals here?

Reply:
Yeah, without a message it'd be a pretty pointless update--oops